test/e2e/Core_update.spec.js

```diff
@@ -654,4 +654,59 @@ describe('Core_updateSettings', () => {
 
     expect(Object.keys(updatedSetting)).toEqual(Object.keys(newSettings));
   });
+
+  it('should update className accordingly', () => {
+    handsontable({
+      data: [[1, true]],
+      className: ['class-1', 'class-2'],
+    });
+
+    const container = spec().$container[0];
+
+    expect(container.classList.contains('class-1')).toBe(true);
+    expect(container.classList.contains('class-2')).toBe(true);
+
+    updateSettings({ className: ['class-1'] });
+
+    expect(container.classList.contains('class-1')).toBe(true);
+    expect(container.classList.contains('class-2')).toBe(false);
+  });
+
+  it('should update tableClassName accordingly', () => {
+    handsontable({
+      data: [[1, true]],
+      tableClassName: ['table_red'],
+    });
+
+    const table = spec().$container.find('table')[0];
+
+    expect(table.classList.contains('table_red')).toBe(true);
+
+    updateSettings({ tableClassName: ['table_green'] });
+
+    expect(table.classList.contains('table_red')).toBe(false);
+    expect(table.classList.contains('table_green')).toBe(true);
+  });
+
+  it('should update className and tableClassName at the same time accordingly', () => {
```

Comment:
I know it's `feature-fix` 😄 although, I don't see any reason to add tests to test them both at the same time. Moreover, it's crucial to test `className` property propagation to cellMeta. Did you test it after your changes?